### ISSUE TITLE
iOS main.jsbundle will not create automatically or update without manual react-native bundle

### ISSUE THREAD

**Issue by nickmccally** [facebook/react-native#18472]
- [x] I have reviewed the [documentation](https://facebook.github.io/react-native)
- [X] I have searched [existing issues](https://github.com/facebook/react-native/issues)
- [X] I am using the [latest React Native version](https://github.com/facebook/react-native/releases)

React native is not creating a main.jsbundle for ios automatically. One must manually be created, even though the scheme set as release.  The code used to create a bundle manually:
`react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios`

**Update** this exact issue is happening on android now for us as well during our Jenkins build process. We are able to manually bundle using: `react-native bundle --entry-file index.js --platform android --dev false --bundle-output android/main.jsbundle --assets-dest android`

No errors occur when this is run, and a bundle is created/updated. From this point forward no bundle updates occur. The following is in **Bundle React Native code and images**:
`export NODE_BINARY=node
../node_modules/react-native/scripts/react-native-xcode.sh`

## Environment

Environment:
  OS: macOS High Sierra 10.13.3
  Node: 8.9.4
  Yarn: 1.3.2
  npm: 5.7.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 3.0 AI-171.4443003

Packages: (wanted => installed)
  react: ^16.3.0-alpha.1 => 16.3.0-alpha.2
  react-native: 0.54.2 => 0.54.2

## Steps to Reproduce

I have been experiencing this with an existing project. I even created a new react native project, and opened the xcode project. It shows main.jsbundle as a missing file. I modified the project scheme to Release, and from there -> Build. No main.jsbundle was created in the iOS project folder. One can manually be created with the code above without an error message.

## Expected Behavior

A new main.jsbundle would update on build when the following is run: build from xcode OR `react-native run-ios --configuration Release` 

## Actual Behavior

No main.jsbundle is created/updated

**Comment by nickmccally** [comment]
We still have to manually create our bundle for iOS, Android bundle is working fine.

**Comment by gorjuspixels** [comment]
We had the same issue, the file was checked into the iOS project. We had to remove it & add a new build step `Bundle React Native code and images`:

```
export NODE_BINARY=node
../node_modules/react-native/scripts/react-native-xcode.sh
```

**Comment by jsina** [comment]
@m-vdb I couldn't realize your workaround. in fact my problem is when I run app on simulator with command `react-native run-ios` everything works fine but when I try to run the app on real device first I have to create the bundle, because bundle not created automatically and even after creating bundle with command `react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios' --assets-dest='./ios'` the static images not shows on ios, although everything works fine in Android devices.

**Comment by m-vdb** [comment]
@jsina you must have another problem that prevents the automatic bundling. Try to debug the execution of `react-native-xcode.sh` to see what is the problem exactly. Maybe you are missing some ios setup with an external library or something like that. Sadly, the same symptom (bundle not created automatically) has a multitude of possible causes.

**Comment by KouSyurei** [comment]
我只想知道为什么明明需要main.jsbundle文件，框架却不自动生成。。。。。

**Comment by nickmccally** [comment]
It is my opinion that this should not be such a widespread issue, and there should be solid documentation to resolving this. I will report back when I have resolved our issue, but currently our CI/CD builds are manually running the bundle script. I've read a bit about some having the wrong path to react-native-xcode.sh, but this has never been the case with our project.

**Comment by nickmccally** [comment]
Update - our CI/CD process is now reverting to an old version of the main.jsbundle and requires a separate command to create the bundle. We are using `react-native bundle --entry-file index.js --platform android --dev false --bundle-output android/main.jsbundle --assets-dest android` in this build process to resolve for it at the moment. We will review further and see why bundling has broken on android as well and report back. This may save some folks research time, as it was a bit hidden on our end.

**Comment by jsina** [comment]
My problem is I've install node with [fishshell](https://fishshell.com/) 
I've change the `Bundle React Native code and image` on `Build Phases` to:
```
export NODE_BINARY=$(fish -c 'which node')
../node_modules/react-native/scripts/react-native-xcode.sh
```
then run `react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios' --assets-dest='./ios'` 
now actually I have images on real device.

**Comment by mjmasn** [comment]
We have the same issue building an android app - building 4 or 5 build types with `./gradlew assembleX` where X is dev/int/beta/release etc and often some or all of the builds will have the wrong js bundle (an old one magically reappears from the depths of some cache) or the APK will be e.g. 15MB instead of the usual 40MB (presumably by having no js bundle at all) and just crash on start. I've tried many things starting with running `./gradlew clean` before each build and using `--reset-cache` but nothing seems to help. Even deleting the two `build` directories, running `watchman watch-del-all` and deleting everything in `/tmp` seems to have no effect. It's becoming somewhat infuriating! 

It appears the bundler is working correctly, as in the generated `js` and `map` files are correct but somehow the APK includes an old bundle (from god knows where) or possible none at all (when we have the APKs that crash instantly).

Build issues like this have now consumed ~100 hours of my life since starting work with RN at the start of the year so I'm dying for a way that will guarantee a 100% fresh build of our dev/int/beta/release builds and for it to be reproducible! If I repeat the build enough times it (usually) eventually works but the fact that it can give completely different output without any code being changed is worrying.

Since 0.56 at least running in dev mode locally seems to have mostly been fixed (just the async storage issue once every 20 or so refreshes) but building APKs remains a big problem (possibly even slightly worse since 0.56 but that's just a gut feeling, I don't have any concrete data).

**Comment by mjmasn** [comment]
Just had the APK size issue again where the first build succeeded and the other 3 failed. From diffing the contents of two of the APKs it looks like none of the assets (images) were copied to the failed one. So that's probably a separate issue to the out of date bundle issue...

**Comment by MrHubble** [comment]
@m-vdb where do you see the output of `react-native-xcode.sh`?

Reference:

>I managed to find out by adding an echo command in react-native-xcode.sh to print the bundling command, and I actually saw the error.

**Comment by m-vdb** [comment]
in the terminal when you build. You can even call the command yourself with the right arguments

**Comment by MrHubble** [comment]
Thanks for taking the time to reply. I was building from Xcode and didn't see any of my `echo` statements in the Xcode console, nor in the terminal that opens for the metro package bundler. Maybe I should have been using `react-native run-ios --configuration Release` from the terminal and I would have seen the output.

**Comment by amirpervaiz086** [comment]
@nickmccally still having the same issue. Using RN 0.57 latest one. I found no main.jsbundle was created so I had to create it manually and after that If any change is made in app.js, it does not reflect in the build unless I create bundle again through the command.

**Comment by shinriyo** [comment]
@amirpervaiz086 me too. `main.jsbundle` didn't create.

**Comment by shinriyo** [comment]
how to make `main.jsbundle`?
I have it `node_modules/react-native/scripts/react-native-xcode.sh`.

**Comment by amirpervaiz086** [comment]
run this command inside project:
 react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios

**Comment by amirpervaiz086** [comment]
Also make sure it is added in copy bundle resources also change this in AppDelegate.m 
jsCodeLocation = [[NSBundle mainBundle] URLForResource:@“main” withExtension:@“jsbundle”];

**Comment by npalethorpe** [comment]
Side note on this issue - I was in the situation where the main.jsbundle wasn't being auto generated on a release build, meaning each time I made a change to my JS I had to re-run the `react-native bundle` script prior to playing in release (a complete headache to do).

Turns out there is a tick box **_"Run script only when installing"_** within Xcode found in '_Build Phases_' > '_Bundle React Native code and images_' section. I have now **unticked** this.

I must have ticked this to fix some other issue and forgotten about it! But now the main.jsbundle script is getting rebuilt each time I run through Xcode. Win.

**Comment by FrenchMajesty** [comment]
In response to @npalethorpe 's solution: Mine was *unticked* and a bundle was still not being built when the schema was Release.

**Comment by ObidosDev** [comment]
Hi all. We had almost the same problem. We have spend around 5h. And at some point we realize that **xCode** use old **node** version. _LOOOL, but excuse me WTF_. Do not ask how and why... I just add `$NODE_BINARY -v` to `../scripts/react-native-xcode.sh` and there was **v7.10.0**. So we close all apps and just set via nvm normal node version (`nvm alias default 10.14.1`). Also u may restart your computer. So the problem have gone for now. Maybe this will be useful for someone else :)

**Comment by foxundermoon** [comment]
```
  "scripts": {
    ...
    "bundle:ios": "react-native bundle   --entry-file   index.js  --bundle-output ios/bundle/index.jsbundle   --dev  false  --platform ios  --assets-dest  ios",
    "postinstall": "npm run bundle:ios"
  }
```

**Comment by coryb08** [comment]
in response to @amirpervaiz086 's solution, be sure to restart xcode

**Comment by vin7974** [comment]
> @shinriyo run this command inside project:
> react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios

I'm getting this error.

`Loading dependency graph, done.

App.js: Cannot read property 'bindings' of null

TypeError: Cannot read property 'bindings' of null`

Any workaround for this issue? my "babel-preset-react-native": "^2.1.0. without upgrading??
Xcode 10, React Native: 0.57.0

**Comment by carlezzo** [comment]
This works for me and generate the **main.jsbundle** but when I change something in code, it does not reload.

When I try to enable Debug JS Remotely throw some error 

`react-native bundle --entry-file index.js --platform ios --bundle-output ios/main.jsbundle --assets-dest ios`

![Simulator Screen Shot - iPhone 8 - 2019-03-14 at 09 52 08](https://user-images.githubusercontent.com/11723151/54358341-e5dea080-463e-11e9-87f3-6c5b20689d68.png)

**Comment by Hitabis** [comment]
In my case I was using a different node version for command line. 
I had
`export PATH="/usr/local/opt/node@8/bin:$PATH"`
in my .bash_profile and xcode did not know about this.

using normal node version fixed it for me.

**Comment by CareerFairPlus** [comment]
I experienced the same issue but it was due to nvm default version being set to a different version than what is defined in the project folder's .nvmrc file.  I would think that this script should honor the version defined in .nvmrc.

**Comment by jpwalton** [comment]
It's been a year since this issue had activity. Anything new?

For the last 6 months, our app has been automatically bundling just fine on both platforms. Now suddenly I have to manually create the bundle on both platforms

**Comment by marcusleeeugene** [comment]
> It's been a year since this issue had activity. Anything new?
> 
> For the last 6 months, our app has been automatically bundling just fine on both platforms. Now suddenly I have to manually create the bundle on both platforms

I came across the same issue as well today.. It used to automatically update bundles for me, and it just suddenly doesn't. Have you found a solution?

**Comment by davidck** [comment]
Just ran into this as well after manually upgrade to RN 0.60.3

**Comment by VenomApps** [comment]
Facing same issue deleted js.mainbundle from all places i knew of reset cache etc. Still it is somehow using an old version of the js

**Comment by marcusleeeugene** [comment]
Hey @VenomApps, here are the solutions that worked for me. First you have to manually create the first bundle file. After that here's what worked for me.

IOS (Add to scripts in package.json file)
`"build:ios": "react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'"`

Android (app/build.gradle file edit project.ext.react to this) 
```
project.ext.react = [
    entryFile: "index.js",
    bundleAssetName: "index.android.bundle",
    bundleInDebug: true,
    bundleInRelease: true
]
```

Hope it works for you!

**Comment by tylermakin** [comment]
I was able to generate the missing `main.jsbundle` file by running `react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'`, but I realized that I had mistakenly tried to double-install node by following the "Installing Dependencies" step of the [React Native CLI Quickstart](https://facebook.github.io/react-native/docs/getting-started).

When I ran `brew install node`, per the Quickstart, it broke the existing (non-homebrew) install of node on my computer and all `npm` commands stopped working. **After [uninstalling and re-installing node](https://stackoverflow.com/a/45348862/3305145), `main.jsbundle` is generated automatically again!** 

```
$ sudo rm -rf /usr/local/lib/node_modules/npm
$ brew reinstall node
```

**Comment by cspicuzza** [comment]
main.jsbundle decided to stop generating during fastlane builds.  I updated to Xcode 11.0 last night.  I'm unsure if this is the culprit.

I've tried the following:
- removing any extra node installs that may exist
- setting the nvm default version to match the project .nvmrc
- @npalethorpe 's solution

**Comment by jaumevn** [comment]
I exactly have the same issue @cspicuzza . Did you found any solution?

**Comment by cspicuzza** [comment]
@jaumevn My issue turned out to be an error in the babel.config

**Comment by jaumevn** [comment]
> @jaumevn My issue turned out to be an error in the babel.config

@cspicuzza what was the error? I'm still having the issue and can find anything on Internet...

**Comment by jaumevn** [comment]
I've finally made it work by using the solution proposed here https://github.com/facebook/react-native/issues/25522#issuecomment-509036251

**Comment by skatea09** [comment]
So after trying to manually install the main.jsbundle file using `react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'` I would get an error about the scheduler not being found in the haste module map. That led me to believe that the scheduler was the culprit.

React native for some reason was not installing the scheduler, even after removing the node_modules and re-installing. After simply running `npm i scheduler` it fixed my main.jsbundle.

**Comment by karthiganesan90** [comment]
> 我只想知道为什么明明需要main.jsbundle文件，框架却不自动生成。。。。。

Google translated to English -

>   I just want to know why the main.jsbundle file is needed, but the framework is not generated automatically. . . . .

**Comment by vedant-dobby** [comment]
> @shinriyo run this command inside project:
> react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios

It's ingenious. There was an error in my code and due to that my bundle wasn't being generated. Thanks a ton for this!!

**Comment by nihp** [comment]
After adding main.jsbundle in Resources my app get crashing in real device

**Comment by tusharpandey** [comment]
> Also make sure it is added in copy bundle resources also change this in AppDelegate.m
> jsCodeLocation = [[NSBundle mainBundle] URLForResource:@“main” withExtension:@“jsbundle”];

, but adding this is not reflecting changes ...

**Comment by almasaeed2010** [comment]
I managed to fix this issue using the following steps:
- If you do not see a file named main.jsbundle in xcode (in the same folder as AppDelegate.m), then you need to run: 
  - ``` react-native bundle --entry-file index.js --platform ios --bundle-output ios/main.jsbundle --assets-dest ios --dev true```
  - Drag and drop the new main.jsbundle from the `YOUR_RN_PROJECT/ios/main.jsbundle` folder into your xcode and make sure to check the copy items checkbox.
- If you see main.jsbundle file but it's empty (as in xcode won't open it), then delete the main.jsbundle in xcode file and follow the steps above.

**Comment by jbromberg** [comment]
I am still having this issue. Everything used to work fine, and then I renamed my Xcode project. After doing this, I had to relink the bundle using this command: `react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'`

Ever since running that command, the bundles are no longer automatically updated.

**Comment by almasaeed2010** [comment]
@jbromberg I managed to fix this by adding `DEBUG=1` to Build Phases -> Preprocessor Macros -> Debug

<img width="706" alt="Screen Shot 2020-06-18 at 1 38 45 PM" src="https://user-images.githubusercontent.com/1512664/85053780-0a41fa80-b169-11ea-9bdf-8a3b47523d33.png">

Notice how `DEBUG=1` shows up twice? If you double click it, and it's not showing then you should add it. Seems to be a glitch of some sort. 

Hope this helps!

**Comment by jbromberg** [comment]
@almasaeed2010  I have checked and I do not have this bug. I am still having the issue unfortunately. I suppose it may be something related to changing the name of my project. The bundling script may be looking in the wrong directory to create the bundle.

**Comment by leinelissen** [comment]
For those still running into issues: you can check the output of the build script in the XCode Report Navigator, all the way at the bottom. For me, when setting up the repository I had renamed the entrypoint from `index.js` to `index.ts`, so I could leverage Typescript for my project. As it turns out, the entrypoint is hardcoded and the build script will refuse to run with another entrypoint. Renaming the entrypoint back to `index.js` solved the issue for me.

TL;DR
**If you have renamed your entrypoint to `index.ts`, change it back to `index.js`.**

**Comment by narayananramu** [comment]
For those still running into issues: check your build _CONFIGURATION_, it should not be **_Debug_**. _react-native-code.sh_ sets `DEV=true` only if it is not _CONFIGURATION_ is not debug.

https://github.com/facebook/react-native/blob/3c9e5f1470c91ff8a161d8e248cf0a73318b1f40/scripts/react-native-xcode.sh#L33-L55

**Comment by mngyuan** [comment]
I generated my own `ios/main.jsbundle` using `react-native bundle` as per some of the suggestions in this thread, and for some reason had it listed in `Copy bundle resources`. Deleting the `ios/main.jsbundle` file and removing it from `Copy bundle resources` got `react-native-xcode.sh` running normally again.

**Comment by NikitaKis** [comment]
This worked for me
https://github.com/facebook/react-native/issues/29351#issuecomment-657929014

**Comment by maxckelly** [comment]
> I generated my own `ios/main.jsbundle` using `react-native bundle` as per some of the suggestions in this thread, and for some reason had it listed in `Copy bundle resources`. Deleting the `ios/main.jsbundle` file and removing it from `Copy bundle resources` got `react-native-xcode.sh` running normally again.

@mngyuan - Did you do it in that order? Delete `ios/main.jsbundle` then remove it from `Copy bundle resources`? I did the same as you but didn't fix it.

**Comment by MasterTeus** [comment]
✅    It worked for me.

Open mac settings> network> select your connection> Advanced> Proxies> Automatic Proxy Discoveries (Uncheck).
![Captura de Tela 2020-11-30 às 17 20 53](https://user-images.githubusercontent.com/39800209/100660207-720f8a80-3330-11eb-8781-853b13033524.png)

**Comment by rci20021028** [comment]
When I run "react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios", I can't see this errors
![image](https://user-images.githubusercontent.com/72734278/101282003-4ef93680-3787-11eb-83b4-64d7e737ff2f.png)

**Comment by ybalaev1** [comment]
> react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios

react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/{**PROJECTNAME**}/main.jsbundle --assets-dest ios

**Comment by udaysagartammina** [comment]
react-native-cli: 2.0.1
react-native: 0.61.5
node: v13.4.0

"build:ios": "react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'"

Run command : npm run build:ios

main.jsbundle file dose not created, created some extra files like **report.20210421.091227.2175.0.001.json**

Any one can help me.

**Comment by ybalaev1** [comment]
replace 
in "build:ios": "react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'"
to  "build:ios": "react-native bundle --entry-file=index.js --bundle-output=./ios/main.jsbundle --dev=false --platform=ios"

**Comment by udaysagartammina** [comment]
> "build:ios": "react-native bundle --entry-file=index.js --bundle-output=./ios/main.jsbundle --dev=false --platform=ios"

➜  Move git:(dev) ✗ npm run build:ios
> Move@0.0.1 build:ios /Users/user/Documents/React_Native/Move
> react-native bundle --entry-file=index.js --bundle-output=./ios/main.jsbundle --dev=false --platform=ios

                 Welcome to React Native!
                Learn once, write anywhere


info Writing bundle output to:, ./ios/main.jsbundle
info Done writing bundle output
warn Assets destination folder is not set, skipping...

Images are not loading in release mode.

**Comment by Osamasomy** [comment]
> 我只想知道为什么明明需要main.jsbundle文件，框架却不自动生成。。。。。

TRANSLATE INTO ENGLISH

I just want to know why the main.jsbundle file is obviously needed, but the framework is not automatically generated

**Comment by robertoclavijob** [comment]
> ```
>   "scripts": {
>     ...
>     "bundle:ios": "react-native bundle   --entry-file   index.js  --bundle-output ios/bundle/index.jsbundle   --dev  false  --platform ios  --assets-dest  ios",
>     "postinstall": "npm run bundle:ios"
>   }
> ```
> 
> change yourself jsbundle file name

This was the only way to get it working on AppCenter. Thanks @foxundermoon

**Comment by peterharding** [comment]
During an upgrade from `0.61 to `0.62`, I encountered a strange state where the packager would start, but would not create the bundle. None of the normal solutions worked for me, and it turns out I had a strange line in my `project.pbxproj` file:

`"GCC_PREPROCESSOR_DEFINITIONS[arch=*]" = "FB_SONARKIT_ENABLED=1";`

Removing this line solved the issue, hopefully this helps someone else save 3 days worth of effort 😓

**Comment by SanjanaTailor** [comment]
> @shinriyo run this command inside project:
> react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios

Thanks a lot this worked for me

**Comment by DmitryBD** [comment]
I was able to fix the bug by changing the project.pbxproj "Build Phases -> Bundle React Native code and images" to
```
export NODE_BINARY=node

# The project root by default is one level up from the ios directory
export PROJECT_ROOT="$PROJECT_DIR"/..

`node --print "require('path').dirname(require.resolve('react-native/package.json')) + '/scripts/react-native-xcode.sh'"`
```
This way all the js bundle and assets will be added while building the code projects. Works with react-native >= 0.63

**Comment by mandotleung-zensis** [comment]
For my case,
- react-native 0.65.1
- use nvm manage the node
- xcode 13
- non M1 mac
- MacOS 12.1
It is found that the shell script issue cause the jsbundle do not generate correctly but the xcode archive task still success.

Change the script in "Build Phases -> Bundle React Native code and images" as below work for me.
```
export RN_DIR="$PROJECT_DIR/../node_modules/react-native" && pwd
export NODE_BINARY=node
../node_modules/react-native/scripts/react-native-xcode.sh
```

**Comment by chellongCoder** [comment]
> My problem is I've install node with [fishshell](https://fishshell.com/) I've change the `Bundle React Native code and image` on `Build Phases` to:
> 
> ```
> export NODE_BINARY=$(fish -c 'which node')
> ../node_modules/react-native/scripts/react-native-xcode.sh
> ```
> 
> then run `react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios' --assets-dest='./ios'` now actually I have images on real device.

your save my day 🎁

**Comment by shayelbaz1** [comment]
> My problem is I've install node with [fishshell](https://fishshell.com/) I've change the `Bundle React Native code and image` on `Build Phases` to:
> 
> ```
> export NODE_BINARY=$(fish -c 'which node')
> ../node_modules/react-native/scripts/react-native-xcode.sh
> ```
> 
> then run `react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios' --assets-dest='./ios'` now actually I have images on real device.

![image](https://user-images.githubusercontent.com/24244828/161557216-6af5003a-1d4b-46f4-8b88-3476def8aef3.png)

**Comment by kennydifiore** [comment]
@shayelbaz1 We had a similar problem when using Node Version Manager (nvm).  If your "default" node installation alias isn't set to the version of node you're using for the RN builds, then it will result in error.  If the build script from react-native included a step to detect if NVM was being used and then call `nvm use` from the main project folder, that would alleviate problems that NVM users have.  The workaround for us was to change the "default" node installation alias in NVM to the one used by our react-native builds.

**Comment by prankymat** [comment]
> @shayelbaz1 We had a similar problem when using Node Version Manager (nvm). If your "default" node installation alias isn't set to the version of node you're using for the RN builds, then it will result in error. If the build script from react-native included a step to detect if NVM was being used and then call `nvm use` from the main project folder, that would alleviate problems that NVM users have. The workaround for us was to change the "default" node installation alias in NVM to the one used by our react-native builds.

Indeed was caused by `nvm` in my case as well. I've removed `.nvm` from my home directory and a few lines in `.zshrc` and fixed it instead.

**Comment by prankymat** [comment]
In my opinion this could have been caught earlier during the `Bundle React Native code and images` phase and throw a compile error instead of letting it continue until the app launches and crashes.

![CleanShot 2022-06-15 at 12 00 26](https://user-images.githubusercontent.com/2998111/173734113-9e61fa2f-d286-4096-84ed-4962f9b2ef08.png)

**Comment by mavine4512** [comment]
run this at the root of the project's terminal: 
react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios' --assets-dest='./ios'

**Comment by saltict** [comment]
> ✅ It worked for me.
> 
> Open mac settings> network> select your connection> Advanced> Proxies> Automatic Proxy Discoveries (Uncheck). ![Captura de Tela 2020-11-30 às 17 20 53](https://user-images.githubusercontent.com/39800209/100660207-720f8a80-3330-11eb-8781-853b13033524.png)

It saved me, I forgot to ignore localhost when config the proxy. Thank you so much!
